### PR TITLE
Port :erlang.list_to_tuple/1 to JS

### LINKED ISSUE #346: Port :erlang.list_to_tuple/1 to JS

### PR DESCRIPTION
**Pull request by rabee361**
Closes #346 

## Summary by CodeRabbit

* **New Features**
  * Added list-to-tuple conversion: proper lists convert to tuples; invalid or improper lists raise an argument error.

* **Tests**
  * Added tests for list-to-tuple (including empty list and error cases).
  * Numerous tests for initialization, OS info, string replace, NFC normalization, and split-related behaviors were removed.

* **Chores / Removals**
  * Removed several string/unicode utilities and init/os helpers; pattern matching/compile behavior reorganized.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: assets/js/erlang/erlang.mjs [hunk 1]
```diff
@@ -1506,6 +1506,18 @@ const Erlang = {
   },
   // End tuple_to_list/1
   // Deps: []
+  // Start list_to_tuple/1
+  "list_to_tuple/1": (list) => {
+    if (!Type.isProperList(list)) {
```
Comment (bartblast) [inline assets/js/erlang/erlang.mjs]: We need to have two separate checks:
1) isList()
2) isProperList()

they have distinct error messages.

File: test/elixir/hologram/ex_js_consistency/erlang/erlang_test.exs [hunk 2]
```diff
@@ -3348,4 +3348,16 @@ defmodule Hologram.ExJsConsistency.Erlang.ErlangTest do
                    {:erlang, :tuple_to_list, [:abc]}
     end
   end
+
+  describe "list_to_tuple/1" do
+    test "returns a tuple corresponding to the given list" do
+      assert :erlang.list_to_tuple([1, 2, 3]) == {1, 2, 3}
+    end
+
+    test "raises ArgumentError if the argument is not a list" do
```
Comment (bartblast) [inline test/elixir/hologram/ex_js_consistency/erlang/erlang_test.exs:3892]: Please add a separate test for an improper list.

File: test/elixir/hologram/ex_js_consistency/erlang/erlang_test.exs [hunk 3]
```diff
@@ -3348,4 +3348,16 @@ defmodule Hologram.ExJsConsistency.Erlang.ErlangTest do
                    {:erlang, :tuple_to_list, [:abc]}
     end
   end
+
+  describe "list_to_tuple/1" do
+    test "returns a tuple corresponding to the given list" do
```
Comment (bartblast) [inline test/elixir/hologram/ex_js_consistency/erlang/erlang_test.exs]: Could you add a test for an empty list?

File: test/javascript/erlang/erlang_test.mjs [hunk 4]
```diff
@@ -5200,4 +5200,26 @@ describe("Erlang", () => {
       );
     });
   });
+
+  describe("list_to_tuple/1", () => {
```
Comment (bartblast) [inline test/javascript/erlang/erlang_test.mjs]: Comments for Elixir tests apply here too.

File: assets/js/erlang/erlang.mjs [hunk 5]
```diff
@@ -1506,6 +1506,18 @@ const Erlang = {
   },
   // End tuple_to_list/1
   // Deps: []
+  // Start list_to_tuple/1
+  "list_to_tuple/1": (list) => {
```
Comment (bartblast) [inline assets/js/erlang/erlang.mjs]: Could you order the function and its tests alphabetically within the files?

**Comment by bartblast** [comment]
Hey @rabee361! Just a heads up - I'm planning to release all the ported functions in a few days. Any chance you'll be able to wrap this one up soon?

**Comment by rabee361** [comment]
> Hey @rabee361! Just a heads up - I'm planning to release all the ported functions in a few days. Any chance you'll be able to wrap this one up soon?

Hey @bartblast 👋 , I apologize for the delay due to work obligations, I'll continue working on this port and it will be finalized in the next 24 hours 👍

**Comment by bartblast** [comment]
@rabee361 Could you pull from dev? It looks like you've got a version that hangs the CI tests.

File: assets/package-lock.json [hunk 6]
```diff
@@ -298,7 +298,6 @@
         }
       ],
       "license": "MIT",
-      "peer": true,
```
Comment (bartblast) [inline assets/package-lock.json:301]: Could you revert changes in package-lock.json?

File: assets/js/erlang/erlang.mjs [hunk 7]
```diff
@@ -1436,6 +1436,25 @@ const Erlang = {
   // End list_to_ref/1
   // Deps: []
 
+  // Start list_to_tuple/1
+  "list_to_tuple/1": (list) => {
+    if (!Type.isList(list)) {
```
Comment (bartblast) [inline assets/js/erlang/erlang.mjs]: Since the errors are the same in the validations we just need `!Type.isProperList(list)` check.

File: test/javascript/erlang/erlang_test.mjs [hunk 8]
```diff
@@ -5558,6 +5558,38 @@ describe("Erlang", () => {
     });
   });
 
+  describe("list_to_tuple/1", () => {
```
Comment (bartblast) [inline test/javascript/erlang/erlang_test.mjs:5561]: We're missing a test case for empty list. The Elixir consistency test has this case, but the JS test doesn't. Please add:

```javascript
it("empty list", () => {
  const result = list_to_tuple(Type.list());

  assert.deepStrictEqual(result, Type.tuple());
});
```

**Comment by bartblast** [comment]
@rabee361 Could you run the formatter: `mix f`?

**Comment by bartblast** [comment]
Hi @rabee361,

This PR has an issue: it shows deletions of code from other recently merged PRs. This happened because merge conflicts were resolved by keeping your old code instead of the new code from `dev`.

**Two ways to fix this:**

1. **Fresh branch (easiest):** Create a new branch from current `dev`, copy over just your `list_to_tuple` changes (the function in `erlang.mjs` and the tests), and open a new PR.

2. **Reset and rebase:** Reset your branch to before the bad merge, then rebase onto current `dev`:
   ```
   git fetch upstream
   git reset --hard <commit-before-merge>
   git rebase upstream/dev
   ```

Let me know if you need help.

**Comment by rabee361** [comment]
> Hi @rabee361,
> 
> This PR has an issue: it shows deletions of code from other recently merged PRs. This happened because merge conflicts were resolved by keeping your old code instead of the new code from `dev`.
> 
> **Two ways to fix this:**
> 
> 1. **Fresh branch (easiest):** Create a new branch from current `dev`, copy over just your `list_to_tuple` changes (the function in `erlang.mjs` and the tests), and open a new PR.
> 2. **Reset and rebase:** Reset your branch to before the bad merge, then rebase onto current `dev`:
>    ```
>    git fetch upstream
>    git reset --hard <commit-before-merge>
>    git rebase upstream/dev
>    ```
> 
> Let me know if you need help.

**I apologize** about that and for taking too long on this port. I created a new PR #623  @bartblast

**Comment by bartblast** [comment]
Closing this PR in favor of #623, which supersedes this one.
